products/llm_analytics/frontend/ConversationDisplay/ConversationMessagesDisplay.tsx

```diff
@@ -231,6 +241,116 @@ export const ImageMessageDisplay = ({
     return <span>{content}</span>
 }
 
+function renderContentItem(item: any, searchQuery?: string): JSX.Element | null {
+    if (typeof item === 'string') {
+        return searchQuery?.trim() ? (
+            <SearchHighlight string={item} substring={searchQuery} className="whitespace-pre-wrap" />
+        ) : (
+            <span className="whitespace-pre-wrap">{item}</span>
+        )
+    }
+
+    if (!item || typeof item !== 'object' || !('type' in item)) {
+        return <HighlightedJSONViewer src={item} name={null} collapsed={5} searchQuery={searchQuery} />
+    }
+
+    if (item.type === 'text' && 'text' in item) {
+        return searchQuery?.trim() && typeof item.text === 'string' ? (
+            <SearchHighlight string={item.text} substring={searchQuery} className="whitespace-pre-wrap" />
+        ) : (
+            <span className="whitespace-pre-wrap">{item.text}</span>
+        )
+    }
+
+    if (item.type === 'image' && 'image' in item && typeof item.image === 'string') {
+        return <ImageMessageDisplay message={{ content: { type: 'image', image: item.image } }} />
+    }
+
+    if (isOpenAIImageURLMessage(item)) {
+        return <img src={item.image_url.url} alt="Message content" className="max-w-full max-h-[400px] rounded" />
+    }
+
+    if (isAnthropicImageMessage(item)) {
+        return (
+            <img
+                src={`data:${item.source.media_type};base64,${item.source.data}`}
+                alt="Message content"
+                className="max-w-full max-h-[400px] rounded"
+            />
+        )
+    }
+
+    if (isGeminiImageMessage(item)) {
+        return (
+            <img
+                src={`data:${item.inline_data.mime_type};base64,${item.inline_data.data}`}
+                alt="Message content"
+                className="max-w-full max-h-[400px] rounded"
+            />
+        )
+    }
+
+    if (isOpenAIFileMessage(item)) {
+        return (
+            // eslint-disable-next-line react/forbid-elements
+            <a href={item.file.file_data} download={item.file.filename} className="text-link hover:underline">
```

Comment:
just checking this one also about if/how paranoid we need to be here about below:

```
Potential XSS concern: if file_data isn't actually a data URL (e.g., javascript:...), this could be exploited. Consider validating it starts with data: before rendering:

  if (!item.file.file_data.startsWith('data:')) {
      return null // or render filename without link
  }
```

eg whats the worse thing a user could upload here and is there any potential attack surface here at all that we need to be extra careful about?